src/jimgw/samplers/blackjax/smc.py

```
@@ -0,0 +1,500 @@
+    )  # persistent adaptive only: per-step covariance scale
+    _tempering_schedule: np.ndarray  # adaptive tempered only: per-step temperature
+    _scan_infos: Any  # fixed ladder: stacked SMCInfo from jax.lax.scan
+
+    def __init__(
+        self,
+        *,
+        n_dims: int,
+        log_prior_fn: Callable,
+        log_likelihood_fn: Callable,
+        log_posterior_fn: Callable,
+        config: BlackJAXSMCConfig = BlackJAXSMCConfig(),
```

Ruff (0.15.12)

[warning] Do not perform function call `BlackJAXSMCConfig` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)